### PR TITLE
Add particle tests and fix nan checks bug

### PR DESCRIPTION
**Pull request by Hendrik1704**
This PR fixes some bugs in the Particle class. Today I wanted to complete the tests for the class and I realized, that we tested in all functions with `self.x == np.nan`, which should be `np.isnan(self.x)`.
I fixed this now in this PR and added test functions for all Particle class functions.

@NGoetz @nilssass Do we want to release a 1.1.1 version to fix this?

### CODE REVIEW

**Comment by Hendrik1704** [comment]
I opened issue #174 to document one of the bugs I solved with the latest push. This is now also reflected in the issue number in the CHANGELOG.

File: src/sparkx/Particle.py [hunk 1]
```diff
@@ -723,7 +723,7 @@ def pdg_valid(self):
 
         Returns
         -------
-        pdg_valid : int
+        pdg_valid : short
```
Comment (nilssass) [inline src/sparkx/Particle.py:726]: Maybe a stupid question: why is the return type short instead of bool?

File: src/sparkx/Particle.py [hunk 2]
```diff
@@ -506,7 +506,7 @@ def pdg(self):
         -------
         pdg : int
         """
-        return self.data_[9]
+        return int(self.data_[9])
```
Comment (nilssass) [inline src/sparkx/Particle.py:509]: Is the casting to int really necessary? Aren't data types set with initialization?
If this is the case, this cast would be obsolete and I would rather not do it as it consumes computational time

**Review by nilssass — APPROVED** [review]
I added two comments with some suggestions but besides these everything looks very good. Thank you for the effort! I already approve the PR as the eventual adjustments are minor

File: src/sparkx/Particle.py (hunk 2 above)
Comment (Hendrik1704) [reply src/sparkx/Particle.py:509]: This seems to be needed. I tried without the cast and it returned a 211.0 for a pion.

File: src/sparkx/Particle.py (hunk 1 above)
Comment (Hendrik1704) [reply src/sparkx/Particle.py:726]: I'll change that to bool.

**Pull request merged**